tests/test_assign_gvi_to_points.py

```diff
@@ -0,0 +1,16 @@
+from typer.testing import CliRunner
+
+from src.assign_gvi_to_points import app
+
+runner = CliRunner(mix_stderr=False)
+
+
+def test_help():
+    """Test the CLI with --help flag."""
+    result = runner.invoke(app, ["--help"])
```

Comment:
Based on how I understand this, does this run the CLI command as if someone really issued the command in CLI? If so, I would consider this more of an integration test as it tests our usage of our dependencies and the CLI command output as well. 

A unit test to me would be calling `main()` directly and mocking any imports. Are we able to mock imports and method parameters this way with the CliRunner? Integration tests are fine, but they test dependencies as well, and I'm not too sure if we can mock stuff with the CliRunner if we want to so we'll have to create test data.

Reply:
Yes, this will run the CLI command as if someone really used it. The particular tests I wrote have no dependencies (it's just printing the help documentation, not running any real data processing). If testing the actual command that does the data processing, I agree we want to run on test data and mock external dependencies (like Mapillary). 

Using the `CliRunner` is still the way you would want to invoke `main` for unit tests (with appropriate mocking). You're generally not meant to run typer-decorated functions as normal functions, and the main way we expect users to invoke them is to use the CLI. 

Typer's testing documentation is here: https://typer.tiangolo.com/tutorial/testing/

Reply:
It looks like running a command via CliRunner does allow for mocking (Typer uses Click): https://stackoverflow.com/questions/39183741/how-to-mock-components-within-unit-tests-for-click-based-cli-applications

So we should be good to go.